soroban-sdk-auth/src/public_types.rs

```diff
@@ -0,0 +1,58 @@
+use soroban_sdk::{contracttype, Bytes, BytesN, Env, EnvVal, Symbol, Vec};
+
+#[derive(Clone)]
+#[contracttype]
+pub struct KeyedEd25519Signature {
+    pub public_key: BytesN<32>,
+    pub signature: BytesN<64>,
+}
+
+pub type AccountAuthorization = Vec<KeyedEd25519Signature>;
+
+#[derive(Clone)]
+#[contracttype]
+pub struct KeyedAccountAuthorization {
+    pub public_key: BytesN<32>,
+    pub signatures: AccountAuthorization,
+}
+
+#[derive(Clone)]
+#[contracttype]
+pub enum KeyedAuthorization {
+    Contract,
+    Ed25519(KeyedEd25519Signature),
+    Account(KeyedAccountAuthorization),
+}
+
+impl KeyedAuthorization {
```

Comment:
I think we should rename `KeyedAuthorization` to `Signature`.

The `Keyed` is a barrier to understanding how authorization works as it is not a concept easily understood only by looking at the name. As far as I can tell we don't provide non-keyed authorization in this contract yet so I don't think there's a consequence to removing it. 

The rename from `Authorization` to `Signature` was discussed [here](https://discord.com/channels/897514728459468821/1011705055130427532/1011713755824005230), and is consistent with the classic protocol where all forms of authorization are described as signatures.

Reply:
See https://github.com/stellar/rs-soroban-sdk/pull/475